src/App.tsx

```diff
@@ -1,27 +1,106 @@
-import { render } from '@testing-library/react';
-import React from 'react';
-import './App.css';
-import Contacts from './Contacts';
-import Header from './Header';
-import Sidebar from './Sidebar';
-import Feed from './Feed';
+import React, { useEffect, useState } from "react";
+import "./App.css";
+import Contacts from "./Contacts";
+import Header from "./Header";
+import Sidebar from "./Sidebar";
+import Feed from "./Feed";
+import CreatePostWindow from "./CreatePostWindow";
 
 function App() {
+  let username = "Meir";
+
+
+  
+  const [postData, setPostData] = useState([""]);
+  const addPost = (input: string) => {
+    setPostData([input, ...postData]);
+  };
+
+  const [isNewPostWindowAlive, setWindowState] = useState(false);
+  const toggleWindow = () => {
+    if (isNewPostWindowAlive && isNewPostInProcess) setGlobalTemp(""); //dispach input from createPostWindow to MessageSenderPlaceholder
```

Comment:
I did not understand what is this globalTemp

Reply:
Local temp stores the text needs to be shown on **Create post window**
Global temp stores the text that needs to be shown outside of it (on **Message Sender**). The main issue is that global temp has to be updated on very specific conditions (and that's the reason it exists).